### ISSUE TITLE
Basic Warning System

### ISSUE THREAD

**Issue by Theoneflop** [Anuken/Mindustry-Suggestions#3038]
### Describe the content or mechanics you are proposing.

The mechanic I'm proposing is a warning system that would warn players whenever some specific things might happen.

Basically, whenever one of these happen, a warning should be shown to the players:

 • Whenever a thorium reactor is being built above a core / whenever a thorium reactor that is above a core is running low on cryo.

 • Whenever a player is doing something that might disrupt the entire power (ex. destroying cryo of an impact that would, if powered off, make the power go negative. counter example: Whilst destroying a thorium reactor that was anyway gonna expload cause of lack of cryo unrelated to the player's past action and the player goes to destroy it should not trigger the warning.)

 • Whenever a player destroys defenses while an enemy is attacking without replacing them with something else after 10 seconds(unless they are logically unable to or if they did so it would just change nothing. Ex: If there are no resources/ if the defenses he might be able to build might not affect the attacker)

 • Whenever a player is unloading resources from the core right into an incinerator while this resource is not close to max.

 • Whenever a player disables a ton of power nodes for the last 20 seconds.

 • Whenever a player is illogically downgrading equipment. (a.k.a. downgrading from airblast to mechanical drill while there is a ton of power and the mined resource is very few inside the core)

 • Whenever a player changes the output of a troop factory to none while it was programmed unto making a specific troop and lacks no resources in doing so, and the player leaves it like that for an entire minute.

### Describe how you think this content will improve the game. If you're proposing new content, mention how it may add more gameplay options or how it will fill a new niche.

This will just make players more aware of their fellow's actions.

### Before making this issue, check the boxes below to confirm that you have acknowledged them.

- [X] I have checked the [Trello](https://trello.com/b/aE2tcUwF/mindustry-trello) to make sure my suggestion isn't planned or implemented in a development version.
- [X] I am familiar with all the content already in the game or have glanced at the wiki to make sure my suggestion doesn't exist in the game yet.
- [X] I have read [README.md](https://github.com/Anuken/Mindustry-Suggestions/blob/master/README.md) to make sure my idea is not listed under the "A few things you shouldn't suggest" category.

**Comment by begginerr** [comment]
i am not a professional but i think that would make the game a lot more laggier

**Comment by begginerr** [comment]
i think a much better way of doing this is just by telling you what your pal is doing at what coordinates, like [name] has built a thorium reactor at 137, 295, or [name] has set [unit factory] to [unit/null]

**Comment by Kyllingene** [comment]
I'm no expert programmer, but I'm pretty sure tracing a cryo producer that is being destroyed down to its destination (of which they could be many), seeing if that destination is an impact, seeing if the cryo is a significant portion of the cryo for that impact, seeing if that impact has any other significant cryo sources, and finally telling you "hey, stop" when it's already too late is not exactly a cakewalk to implement. Just don't be dumb.

It would also be quite difficult to detect if a thorium reactor is exploding due to consistent lack of cryo, then stopping that warning in that case, but also making sure that it would not get more cryo soon anyways (since otherwise this would be a super annoying warning).

It would also be extremely difficult to detect if something is illogical, and to be functional would require a ton of different factors that would all have to be manually checked.

It would also be extremely difficult to detect if defenses are unnecessary at the current moment, and if them being replaced would be useful or not, also requiring a ton of different factors that would all have to be manually checked.

Basically, it would be a matter of hardcoding a thousand different edge cases (and still not even scratching the surface of the possible cases), and then spending a few months getting some of the bugs fixed (you would never have them all fixed).

**Comment by Feilong-Mindustry** [comment]
Beside the difficulties of implementing the suggested feature, it also takes away from the game. Making mistakes and learning the consequences is a part of experience. There is a path of growth for a new player: learning basics, unlocking tech tree, understanding efficient design, making mistakes, implementing mlog. Do we really need to warn new players about every mistake they can make? I don't think so.

**Comment by EvilBadger34** [comment]
I think an alarm block would be better, which could be connected by logic blocks that monitor things you want them to

**Comment by buthed010203** [comment]
Lots of these things already exist in mods and clients anyways
